### PR TITLE
 Update naming for AdClient

### PR DESCRIPTION
**Pull request by taquitos**
Apple might have updated their app scanning to find any mention of iAd, rejecting apps even though it isn't used.

### CODE REVIEW

File: IntegrationTests/XcodeDirectIntegration/XcodeDirectIntegration.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -548,7 +578,7 @@
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = com.revenuecat.XcodeDirectIntegration;
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				PROVISIONING_PROFILE_SPECIFIER = "match AppStore com.revenuecat.XcodeDirectIntegration 1604090134";
```
Comment (taquitos) [inline IntegrationTests/XcodeDirectIntegration/XcodeDirectIntegration.xcodeproj/project.pbxproj:551]: New certs

File: Purchases/Attribution/RCAttributionFetcher.h [hunk 2]
```diff
@@ -29,7 +29,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (nullable NSString *)identifierForVendor;
 
-- (void)adClientAttributionDetailsWithCompletionBlock:(RCAttributionDetailsBlock)completionHandler;
+- (void)afficheClientAttributionDetailsWithCompletionBlock:(RCAttributionDetailsBlock)completionHandler;
```
Comment (beylmk) [inline Purchases/Attribution/RCAttributionFetcher.h:32]: why affiche? affiche = attach in french? wondering how we can document this while avoiding the-words-that-must-not-be-named.

or maybe we could do a funny misspelling like  `AhdClyent`?

File: Purchases/Attribution/RCAttributionTypeFactory.m [hunk 3]
```diff
@@ -29,35 +30,35 @@ - (NSString *)mangledTrackingClassName {
...
 
-    return (Class<FakeTrackingManager> _Nullable)NSClassFromString(className);
+    return (Class<FakeFollowingManager> _Nullable)NSClassFromString(className);
 }
 
-- (Class<FakeASIdentifierManager> _Nullable)asIdentifierClass {
+- (Class<FakeASIdManager> _Nullable)asIdClass {
     // We need to do this mangling to avoid Kid apps being rejected for getting idfa.
     // It looks like during the app review process Apple does some string matching looking for
     // functions in the AdSupport.framework. We apply rot13 on these functions and classes names
```
Comment (beylmk) [inline Purchases/Attribution/RCAttributionTypeFactory.m:46]: maybe we should remove AdSupport.framework and idfa in the comment? or are we not worried about that?

File: Purchases/Attribution/RCAttributionFetcher.h (hunk 2 above)
Comment (aboedo) [reply Purchases/Attribution/RCAttributionFetcher.h:32]: +1. That'd make it easier for devs to track down the original class. `adKlient`?

File: Purchases/Attribution/RCAttributionTypeFactory.m (hunk 3 above)
Comment (aboedo) [reply Purchases/Attribution/RCAttributionTypeFactory.m:46]: can't hurt

**Review by aboedo — COMMENTED** [review]
I left a comment in the other PR regarding linking to the original classes through a rev.cat link, that applies here as well

File: Purchases/Attribution/RCAttributionTypeFactory.m (hunk 3 above)
Comment (taquitos) [reply Purchases/Attribution/RCAttributionTypeFactory.m:46]: Last time I checked, comments don't get compiled 😄

File: Purchases/Attribution/RCAttributionFetcher.h (hunk 2 above)
Comment (taquitos) [reply Purchases/Attribution/RCAttributionFetcher.h:32]: Lolll good idea
`adKlient` 
I kinda have insight into how Apple does scanning for words (I was part of the China Online Store launch). At least there, they look for permutations of the same word using different letters that have the same sound or are similar. So `AdKlient` would get caught, `@dClient` `@dKlient` as well. The word needs to be different enough from the word they are scanning for. 

This is all just a guess, of course.

Comment (taquitos) [reply Purchases/Attribution/RCAttributionFetcher.h:32]: So actually, I'm pretty sure `AddClyent` would also get caught.

File: Purchases/Attribution/RCAttributionTypeFactory.m (hunk 3 above)
Comment (aboedo) [reply Purchases/Attribution/RCAttributionTypeFactory.m:46]: right, forgot for a minute that the binary that makes it to app store builds won't have the comments in it

File: IntegrationTests/CocoapodsIntegration/CocoapodsIntegration.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -508,7 +508,7 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				ASSETCATALOG_COMPILER_GLOBAL_ACCENT_COLOR_NAME = AccentColor;
 				CLANG_ENABLE_MODULES = YES;
-				CODE_SIGN_IDENTITY = "iPhone Developer";
+				CODE_SIGN_IDENTITY = "iPhone Distribution";
```
[the remark this answers is not in the document]
Comment (aboedo) [reply IntegrationTests/CocoapodsIntegration/CocoapodsIntegration.xcodeproj/project.pbxproj]: oops, clicked resolve when I meant to reply. 
I think this maps the distribution cert to both debug and release, ideally we'd map debug -> development, release -> distribution, but it's not a big deal given that we don't run these directly, just for test archiving

[the remark this answers is not in the document]
Comment (taquitos) [reply IntegrationTests/CocoapodsIntegration/CocoapodsIntegration.xcodeproj/project.pbxproj]: I don't think so, I backed this out.

**Review by aboedo — APPROVED** [review]

**Pull request merged**